block/manager.go

```diff
@@ -794,6 +771,7 @@ func (m *Manager) execCreateBlock(_ context.Context, height uint64, lastSignatur
 	for i := range batchData.Transactions {
 		blockData.Txs[i] = types.Tx(batchData.Transactions[i])
 	}
+	header.DataHash = blockData.Hash()
```

Comment:
DataHash should not be set to `Data.Hash`, instead it should be set to `batch data`. Please read the validation ADR. Data.Hash is used for p2p gossip. For header validation, we must use batch data.

Reply:
we never made the migration as there was a flow issue with it. we should open an issue on the validation to do that implementation